go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -199,6 +199,40 @@ func (oc *BaseSecondaryNetworkController) Reconcile(netInfo util.NetInfo) error
 	return util.ReconcileNetInfo(oc.ReconcilableNetInfo, netInfo)
 }
 
+func (oc *BaseSecondaryNetworkController) FilterResource(objType reflect.Type, obj interface{}) bool {
+	if !oc.IsPrimaryNetwork() {
+		return true
+	}
+	switch objType {
+	case factory.NamespaceType:
+		ns, ok := obj.(*kapi.Namespace)
+		if !ok {
+			klog.Errorf("Failed to cast the provided object to a namespace")
+			return false
+		}
+		namespaceNet, err := oc.networkManager.GetActiveNetworkForNamespace(ns.Name)
```

Comment:
rather than rely on active network for namespace, you can just do oc.GetNamespaces() and check if the namespace is in the list.

Reply:
Good idea, this also addresses https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4861#discussion_r1911318631 as it should work for all secondary networks.